frontend/src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertProps,
+  AlertTitle,
+  Box,
+  CloseButton,
+  useMediaQuery,
+  UseToastOptions,
+} from '@chakra-ui/react'
+
+type ToastStatus = 'danger' | 'success' | 'warning'
+
+export interface ToastProps
+  extends Omit<
+    UseToastOptions,
+    'duration' | 'position' | 'render' | 'status' | 'variant'
+  > {
+  /**
+   * The status of the Toast.
+   */
+  status: ToastStatus
+  // RenderProps that chakra passes to all custom components that uses the render function
+  onClose: () => void
+}
+
+const getChakraAlertStatus = (status: ToastStatus): AlertProps['status'] =>
+  status === 'danger' ? 'error' : status
+
+export const Toast = ({
+  status,
+  title,
+  id,
+  description,
+  isClosable,
+  onClose,
+  onCloseComplete,
+}: ToastProps): JSX.Element => {
+  // Dynamically determine the size of the modal based on screen breakpoints
+  const [isDesktop] = useMediaQuery('(min-width: 1080px)')
+  const width = isDesktop ? '680px' : 'auto'
+  const horizontalMargins = isDesktop ? 'inherit' : 2
+
+  return (
+    <Alert
+      status={getChakraAlertStatus(status)}
+      variant="toast"
+      colorScheme={status}
+      padding={4}
+      id={String(id)}
+      // Padding right is 4 rem (normal padding) + width of the button.
+      // This is to prevent the button overlapping the text on resize.
+      pr={10}
+      width={width}
+      mx={horizontalMargins}
+      mt={2}
+    >
+      <AlertIcon />
+      <Box>
+        {title && <AlertTitle>{title}</AlertTitle>}
+        {description && (
+          <AlertDescription display="block">{description}</AlertDescription>
+        )}
+      </Box>
+      {isClosable && (
+        <CloseButton
+          position="absolute"
```

Comment:
style props should go into theme config.

The close button is also not conforming to the Figma design; icon used should be BoxIcon's close button. I recall there being a prop to replace the icon so should be ez